core/src/main/java/com/amplifyframework/core/category/Category.java

```diff
@@ -61,44 +62,38 @@ public Category() {
      * @param context An Android Context
      * @throws AmplifyException if already configured
      */
-    public final void configure(CategoryConfiguration configuration, Context context)
+    public final void configure(@NonNull CategoryConfiguration configuration, @NonNull Context context)
             throws AmplifyException {
-        if (isConfigured) {
-            throw new AmplifyException("Amplify was already configured",
-                    "Be sure to only call Amplify.configure once");
-        }
+        synchronized (configurationState) {
+            validateConfigurationState(ConfigurationState.NOT_CONFIGURED);
 
-        for (P plugin : getPlugins()) {
-            String pluginKey = plugin.getPluginKey();
-            JSONObject pluginConfig = configuration.getPluginConfig(pluginKey);
+            for (P plugin : getPlugins()) {
+                String pluginKey = plugin.getPluginKey();
+                JSONObject pluginConfig = configuration.getPluginConfig(pluginKey);
+                plugin.configure(pluginConfig, context);
+            }
 
-            plugin.configure(pluginConfig, context);
+            configurationState.set(ConfigurationState.CONFIGURED);
```

Comment:
shouldn't this be set outside the `for` loop?

Reply:
YES! Thanks, will fix.

Reply:
oh wow that was really confusing LOL